mitodl/open-discussions
# Connect LearningResource Drawer to router

#### Pre-Flight checklist


#### What are the relevant tickets?
#3734 

#### What's this PR do?
This PR connects the learning resource drawer to the router. Now when you open or close it, the URL updates to reflect that change.

**Additional changes:**
- the `ExpandedLearningResourceDisplay` runs dropdown is now managed wholly within that component
- a some new tests (for the runs dropdown; for the search page; for the drawer routing)
- restructured the info display in `ExpandedLearningResourceDisplay`

#### How should this be manually tested?
1. Open the learning resource drawer on a field page. The URL should update. Close the drawer. The URL should update.
2. Repeat (1) on the search page. In the developer console, check that no extra search API requests are being made when the drawer opens/closes.
3. View the learning resource drawer for courses and videos. Check that it looks correct. _I made some changes to the internal logic of the drawer; there should be essentially no visible difference._
4. Find a course with multiple runs. Check that you can change the run and that the data display updates correctly.

#### Screenshots (if appropriate)

Drawer, **Before** / **After**:

I made some changes to the implementation of the LearningResourceDrawer. There should be basically no visible change. (*Well, "Info" is a little bigger now. And "Offered by" is gray., which seemed like the original intent.)

<img width="400" alt="Screen Shot 2022-09-23 at 11 44 24 AM" src="https://user-images.githubusercontent.com/9010790/192000535-63d97e9f-aebc-4808-b865-35004ce0540c.png"> <img width="350" alt="Screen Shot 2022-09-23 at 12 00 52 PM" src="https://user-images.githubusercontent.com/9010790/192003555-586f4dff-78c4-40a0-9745-26fac4783b1f.png">

### Where to Start?
I would start with `LearningResourceDrawer.tsx` and `RoutedDrawer.tsx` (and its test file).
